### PR TITLE
Adding hand wired Bento macro pad.

### PR DESCRIPTION
**Pull request by ghostseven**
Using Keebio/bd9n keyboard as a base, adapted and added a new keyboard for a 2x3 macro pad with a single rotary encoder.  Keyboard is described and available [here](https://www.thingiverse.com/thing:4416966), original reference uses the bd9n firmware, the objective is to add a firmware specifically for this keyboard. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ghostseven/bento/info.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "keyboard_name": "Bento Macropad",
+  "url": "https://www.thingiverse.com/thing:4416966",
+  "maintainer": "Ghostseven",
+  "width": 3,
+  "height": 3,
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/info.json:6]:
```suggestion
  "height": 2,
```

File: keyboards/ghostseven/bento/readme.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# BENTO
+
+![BENTO](https://cdn.thingiverse.com/assets/0b/6a/84/9c/9a/large_display_IMG_3374.jpg)
+
+A hand wired 2x3 macropad with a rotary encoder at the upper right corner.
+
+Keyboard Maintainer: [Dwin17](https://www.thingiverse.com/dwin17) - Hardware [GhostSeven](https://github.com/ghostseven) - QMK Firmware   
+Hardware Supported: Pro Micro, Elite-C, Proton C   
+Hardware Availability: [Thingiverse](https://www.thingiverse.com/thing:4416966)
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/readme.md:9]:
```suggestion
* Keyboard Maintainer: [Dwin17](https://www.thingiverse.com/dwin17) - Hardware [GhostSeven](https://github.com/ghostseven) - QMK Firmware
* Hardware Supported: Pro Micro, Elite-C, Proton C
* Hardware Availability: [Thingiverse](https://www.thingiverse.com/thing:4416966)
```

File: keyboards/ghostseven/bento/readme.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ghostseven/bento/rev1:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+
+
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/readme.md:18]:
```suggestion
```

File: keyboards/ghostseven/bento/rev1/config.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCB10
+#define PRODUCT_ID      0x1133
+#define DEVICE_VER      0x0100
+#define MANUFACTURER    GhostSeven
+#define PRODUCT         Bento Macropad Rev. 1
+#define DESCRIPTION     2x3 Macropad with Rotary Encoder
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/config.h:28]:
```suggestion
```

File: keyboards/ghostseven/bento/rev1/rules.mk [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rules.mk:11]:
```suggestion
```

File: keyboards/ghostseven/bento/rev1/rules.mk [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rules.mk:33]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/ghostseven/bento/rev1/rules.mk [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality on B7 by default
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rules.mk:26]:
```suggestion
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
```

File: keyboards/ghostseven/bento/rev1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rules.mk:18]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/ghostseven/bento/rev1/config.h [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCB10
+#define PRODUCT_ID      0x1133
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/config.h:24]:
```suggestion
#define VENDOR_ID       0xFEED
#define PRODUCT_ID      0xDAD3
```
```
echo -n "Bento Macropad Rev. 1" | shasum | cut -c1-4
```

File: keyboards/ghostseven/bento/rev1/rev1.c [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#include "rev1.h"
+
+void eeconfig_init_kb(void) {
+#ifdef BACKLIGHT_ENABLE
+    backlight_enable();
+    backlight_level(5);
+#endif
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rev1.c:7]:
```suggestion
```

File: keyboards/ghostseven/bento/rev1/rev1.c [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+#ifdef RGBLIGHT_ENABLE
+    rgblight_enable(); // Enable RGB by default
+    rgblight_sethsv(0, 255, 255);  // Set default HSV - red hue, full saturation, full brightness
+#ifdef RGBLIGHT_ANIMATIONS
+    rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL + 2); // set to RGB_RAINBOW_SWIRL by default
+#endif
+#endif
+
+    eeconfig_update_kb(0);
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rev1.c:16]:
```suggestion
```

File: keyboards/ghostseven/bento/rev1/config.h [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    { E6, B4, B2 } \
+}
+
+#define ENCODERS_PAD_A { D1 }
+#define ENCODERS_PAD_B { D0 }
+
+#define BACKLIGHT_PIN B5
+#define BACKLIGHT_BREATHING
+#define BACKLIGHT_LEVELS 7
+
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/config.h:45]:
```suggestion
```

File: keyboards/ghostseven/bento/keymaps/default/config.h [hunk 13]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/ghostseven/bento/readme.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+* Hardware Supported: Pro Micro, Elite-C, Proton C
+* Hardware Availability: [Thingiverse](https://www.thingiverse.com/thing:4416966)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ghostseven/bento/rev1:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/readme.md:17]:
```suggestion
```

File: keyboards/ghostseven/bento/rev1/rules.mk [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/rev1/rules.mk:19]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
```

File: keyboards/ghostseven/bento/config.h [hunk 16]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (fauxpark) [inline keyboards/ghostseven/bento/config.h:1]: This file can be deleted.

**Comment by fauxpark** [comment]
Since this is a handwired board, it must also be moved to `keyboards/handwired`.

**Comment by cbc02009** [comment]
@ghostseven Would it be possible to piggy-back a second keymap for bento onto this pull request? I have one I've been using but never committed. It's attached as a text file.

If not, I'll just wait until this is merged in and make a second pull request. 

[keymap.c.txt](https://github.com/qmk/qmk_firmware/files/5057769/keymap.c.txt)

**Comment by ghostseven** [comment]
> @ghostseven Would it be possible to piggy-back a second keymap for bento onto this pull request? I have one I've been using but never committed. It's attached as a text file.
> 
> If not, I'll just wait until this is merged in and make a second pull request.
> 
> [keymap.c.txt](https://github.com/qmk/qmk_firmware/files/5057769/keymap.c.txt)

I don't see why not, the more keymaps the better, added.

File: keyboards/handwired/bento/bento.c [hunk 17]
```diff
@@ -0,0 +1 @@
+#include "bento.h"
```
Comment (noroadsleft) [inline keyboards/handwired/bento/bento.c:1]: Need a license header here.

File: keyboards/handwired/bento/bento.h [hunk 18]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/handwired/bento/bento.h:1]: Need a license header here.

File: keyboards/handwired/bento/rev1/rev1.c [hunk 19]
```diff
@@ -0,0 +1,13 @@
+#include "rev1.h"
```
Comment (noroadsleft) [inline keyboards/handwired/bento/rev1/rev1.c:1]: Need a license header here.

**Pull request merged**